lib/presentation/mixins/common_media_picker_mixin.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:ui';
+
+import 'package:fluffychat/utils/permission_dialog.dart';
+import 'package:fluffychat/utils/permission_service.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:wechat_camera_picker/wechat_camera_picker.dart';
+
+typedef OnSendPhotosTap = void Function()?;
+
+mixin CommonMediaPickerMixin {
+  Future<PermissionStatus>? getCurrentPhotoPermission() {
+    return PermissionHandlerService().requestPermissionForPhotoActions();
+  }
+
+  Future<PermissionStatus>? getCurrentCameraPermission() {
+    return PermissionHandlerService().requestPermissionForCameraActions();
```

Comment:
can we use DI or create instance in Mixin ?

Reply:
what is the DI? Try to minimize the complex of mixin, I think.

Reply:
Dependency Injection
PermissionHandlerService maybe have different implement on Web , Desktop